agent/ui_endpoint.go

```diff
@@ -16,7 +17,9 @@ import (
 const metaExternalSource = "external-source"
 
 type GatewayConfig struct {
-	ListenerPort int
```

Comment:
I'm pretty sure we don't need to keep `ListenerPort` around for backwards compatibility, but I wonder what is the best way of ensuring that the API and UI stay in sync in released versions?

One option is to leave `ListenerPort` in the API response for this PR, and then remove it when the UI updates to use the new `Addresses` field?

Reply:
Maybe I'm missing something but I don't think it's possible for the UI and API to be out of sync. The UI calls out to the agent it was packaged with.

Reply:
Or are you asking about the case where an API change was made but the UI wasn't updated for that release?

Reply:
Yes that case is what I am thinking of. Imagine we merge this PR in 1.8.1, but for some reason the UI change does not make it into that release. By removing `ListenerPort` here, we are requiring UI changes to be made before 1.8.1 is released and need to coordinate that.